projecttacoma/bulk-export-app
# TACOSTRAT 1524 Display Ndjson Data and Enhanced Export Execution Page

# Summary

This PR adds buttons that display the ndjson files to the user before downloading them. It also makes the export execution page more functional and aesthetic.

[JIRA Issue](https://jira.mitre.org/browse/TACOSTRAT-1524)

### Important: 

This is the same section as the previous PR

If you want to simulate the request process with the server taking longer to generate a response, go to line 17 of `exportWorker.js` in the bulk export server and add this line:

`await new Promise(resolve => setTimeout(resolve, 5000));` 

This adds a five second delay to the export worker. Save this and stop and rerun the server.


## New behavior

Navigate to the export execution page as usual, selecting a export level, some type parameters, and clicking the button to continue. The export execution page is now updated, with the polling logs and the requested files section having new styling. There is now a button to the left of the download button, which if you hover over will tell you that you can preview the ndjson data for the  requested file. When you click this button, it will open a modal with the code for a single resource on that file visible.  The numbers at the bottom of the modal let you go through the resources and look at each one individually.


## Code changes

- `package.json` - added mantine code highlight package to display the json data in a  readable way.
- `src/app/export-execution/layout.tsx` changed the header size and style
- `src/app/export-execution/page.tsx` - added the above functionality.
- `src/components/export-execution/file-downloads.tsx` - displays the files that come from the content location along with their filesizes, number of resources, a button to preview, and a button to download the file.
src/app/export-execution/layout.tsx
- `src/app/favicon.ico` added this back to have an icon for the app in the tab.
- `src/app/global.module.css` created this to add the app header styles.
- `src/app/query-builder/layout.tsx` changed this to use the global styles file
- `src/app/query-selector/layout.tsx` same as above
-  `src/components/export-execution/export-status-info.tsx`  changed filename from `polling-logs.tsx` to be more specific. Added styling and a status badge to the header of the section.
- `src/components/export-execution/requested-files.tsx` changed from `file-downloads.tsx` to be more specific. Added styling and button to open the preview modal, along with logic to handle that.
- `src/components/export-execution/resource-file-preview.tsx` content of the file preview modal.
- `src/util/ndjsonParsers.ts` created file to have functions to manipulate ndjson data.


# Testing guidance
- Make sure that the changes to the bulk-export-server are made described in the summary section
- Run the bulk export server
- `npm run check`
- `npm run build`
- `npm run start`
- Navigate through the pages and get to the export execution page.
- Wait for the export, then preview a file. Verify that this process is as described and feels right.
